src/tests/uat/recommendation_uat.py

```diff
@@ -0,0 +1,423 @@
+
+"""
+The following API calls are tested in this script.
+
+For the validation CBC API requests are used.
+
+To execute, a profile must be provided using the standard CBC Credentials.
+
+Recommendations:
+* Search Recommendations
+
+The following API calls were not covered:
+* Recommendation Workflow
+"""
+
+import sys
+import requests
+from cbc_sdk.helpers import build_cli_parser, get_cb_cloud_object
+from cbc_sdk.endpoint_standard import Recommendation
+
+
+def compare(context, object_field, raw_data, raw_field):
+    """
+    Compare an object field value against the raw field value it should have come from.
+
+    Args:
+        context (str): The context of the comparison, used for printing error messages.
+        object_field (Any): The value of the object field being compared.
+        raw_data (dict): Raw data structure we're comparing a field from.
+        raw_field (str): Name of the raw field we're doing the comparison on.
+
+    Returns:
+        bool: True if comparison was successful, False if not.
+    """
+    if raw_field in raw_data:
+        result = (object_field == raw_data[raw_field])
+    else:
+        result = (object_field is None)
+    if not result:
+        print(f"field value {raw_field} did not match in {context} - object value {object_field}, "
+              f"raw value {raw_data.get(raw_field, None)}")
+    return result
+
+
+def check_not_present(context, raw_data, raw_field):
+    """
+    Verifies that a specified field is not present in a raw data structure.
+
+    Args:
+        context (str): The context of the comparison, used for printing error messages.
+        raw_data (dict): Raw data structure we're checking a field from.
+        raw_field (str): Name of the raw field we're verifying is not present.
+
+    Returns:
+        True if the field was not present, False if it was.
+    """
+    if raw_field not in raw_data:
+        return True
+    print(f"field value {raw_field} should be absent from {context}, but is not")
+    return False
+
+
+def compare_impact(context, impact, raw_data):
+    """
+    Compare a RecommendationImpact object against the data that should represent it.
+
+    Args:
+        context (str): The context of the comparison, used for printing error messages.
+        impact (RecommendationImpact): The RecommendationImpact object to compare.
+        raw_data (dict): The dictionary containing what should be the same data.
+
+    Returns:
+        bool: True if comparison was successful, False if not.
+    """
+    result = compare(context, impact.event_count, raw_data, 'event_count')
+    result = compare(context, impact.impact_score, raw_data, 'impact_score') and result
+    result = compare(context, impact.impacted_devices, raw_data, 'impacted_devices') and result
+    result = compare(context, impact.org_adoption, raw_data, 'org_adoption') and result
+    result = compare(context, impact.update_time, raw_data, 'update_time') and result
+    return result
```

Comment:
Can you print a message if 'result' is false before returning it? In this case, the message can be something like 'Comparing impact failed' or similar. Thus, in case of test failures, it would be easier to spot the exact failing step.

Reply:
The `compare()` function does that.

```
 if not result:
        print(f"field value {raw_field} did not match in {context} - object value {object_field}, "
              f"raw value {raw_data.get(raw_field, None)}")
```

Reply:
Thanks! I've somehow missed it.